modules/sc-mesh-secure-deployment/src/1_5/common/test/config_spectralscan.py

```diff
@@ -3,60 +3,84 @@
 import os
 import re
 
-
-# Get mesh interface and frequency channel being used
+# Get mesh interface and current frequency
 iw_output = os.popen('iw dev').read()
 iw_output = re.sub('\s+', ' ', iw_output).split(' ')
 
-
-# Split list for multiple interfaces
+# Parse multiple interfaces
 size = len(iw_output)
 idx_list = [idx - 1 for idx, val in enumerate(iw_output) if val == "Interface"]
 if(len(idx_list) > 1): # if there is more than one interface up, create list for each
-    idx_list.pop(0)
+    idx_list.pop(0) # print info not needed
 iw_interfaces = [iw_output[i: j] for i, j in zip([0] + idx_list, idx_list + ([size] if idx_list[-1] != size else []))]
 
-
-# Check if mesh interface is up, if not exit
+# Check if mesh interface is up and get interface name and current channel
 no_mesh = False
 for interface_list in iw_interfaces:  
     try:
         mesh_index = interface_list.index("mesh")
         mesh_list = interface_list
         no_mesh = False
+        break
     except:
         no_mesh = True
 
 if(no_mesh):    
     sys.exit("No mesh interface to scan/No mesh set up")
-    
 
 mesh_interface_index = mesh_list.index("Interface") + 1
 mesh_interface = mesh_list[mesh_interface_index].split()[0]
 channel_index = mesh_list.index("channel") + 2 # index of channel frequency
 channel_freq = re.sub("[^0-9]", "", mesh_list[channel_index]).split()[0]
 
+# Get channels available to be scanned by interface
+iw_channels = os.popen(f'iwlist {mesh_interface} channel').read()
+iw_channels = re.sub('\s+', ' ', iw_channels).split('GHz')
 
-# Temporary workaround for mesh scanning: mesh interface cannot scan it's own channel so remove it temporarily
-all_bands = '2412 2417 2422 2427 2432 2437 2442 2447 2452 2457 2462 5180 5200 5220 5240 5260 5280 5300 5320 5745 5765 5785 5805 5825'
-all_2_4 = '2412 2417 2422 2427 2432 2437 2442 2447 2452 2457 2462'
-all_5 = '5180 5200 5220 5240 5260 5280 5300 5320 5745 5765 5785 5805 5825'
+channel_list = []
+count = 1
+for channel in iw_channels:
+    try:
+        if(count == 1):
+            temp_list = re.sub('\s+', ' ', channel).split(':')
+            freq_val = int(float(temp_list[-1].strip())*10**3)
+        else:
+            freq_val = int(float(channel[channel.index(":")+1:].strip())*10**3)
+        
+        channel_list.append(freq_val)
+        count += 1
+    except:
+        None
 
+# Temporary workaround for mesh scanning: mesh interface cannot scan it's own channel so remove it temporarily
+all_bands = ' '.join(str(ch) for ch in channel_list)
 all_bands = all_bands.replace(channel_freq, '').strip()
-all_2_4 = all_2_4.replace(channel_freq, '').strip()
-all_5 = all_5.replace(channel_freq, '').strip()
 
+# Get 2_4GHz and 5GHz channels from full list of channels retreived
+all_5 = [5180, 5200, 5220, 5240, 5260, 5280, 5300, 5320, 5745, 5765, 5785, 5805, 5825, 5845] #check if any one of 5ghz channels are in available list of channels to scan
```

Comment:
iw list will give u the list of allowed channel based on regulatory and CC settings, you can parse and populated from there.

Reply:
yes that's being done in lines 37 - 55! It will return a full list of 2.4 and 5 GHz channels that are available to scan, Returned as "channel_list"

The "all_5" list you referenced is just looped through to find in the "channel_list" the first instance of a 5 GHz channel in order to split the full list of 2.4 and 5 into:
- 2.4 band channels
- 5 band channels
separately in the case of a band specific scan

Reply:
I've updated the code in order to split into 2.4 and 5 bands dynamically